### PR TITLE
[mds-db] [mds-policy-author][mds-compliance] Fix/final separation of policy from geo

### PR DESCRIPTION
**Pull request by janedotx**
…s an error.

## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: packages/mds-db/policies.ts [hunk 1]
```diff
@@ -179,19 +179,23 @@ export async function publishPolicy(policy_id: UUID) {
...
+      geographies.map(async geography_id => {
+        const isPublished = await isGeographyPublished(geography_id)
+        if (!isPublished) {
+          return geography_id
+        }
+        return null
       })
     )
 
+    unpublishedGeoIDs.forEach(id => {
```
Comment (macsj200) [inline packages/mds-db/policies.ts]: You could potentially use `filterEmpty()` here to save yourself an `if` statement

File: packages/mds-compliance/tests/api.spec.ts [hunk 2]
```diff
@@ -799,14 +810,12 @@ describe('Tests Compliance API:', () => {
...
-        Promise.all([cache.seed(seedData), db.seed(seedData)]).then(() => {
-          db.writeGeography({ name: 'la', geography_id: GEOGRAPHY_UUID, geography_json: la_city_boundary }).then(() => {
-            db.writePolicy(COUNT_POLICY_JSON_4).then(() => {
-              db.publishPolicy(COUNT_POLICY_JSON_4.policy_id).then(() => {
-                done()
-              })
-            })
-          })
+        Promise.all([cache.seed(seedData), db.seed(seedData)]).then(async () => {
+          await db.writeGeography({ name: 'la', geography_id: GEOGRAPHY_UUID, geography_json: la_city_boundary })
```
Comment (karcass) [inline packages/mds-compliance/tests/api.spec.ts]: can we copy `la_city_boundary` from `ladot_service_areas` into `mds-test-data`?  `ladot_service_areas` is on the chopping block.

Comment (avatarneil) [reply packages/mds-compliance/tests/api.spec.ts]: @karcass Yup. @janedotx let's keep this PR open until my service-area purging PR lands, then it'll be much easier for you to implement the requested change :)

File: packages/mds-policy-author/api.ts [hunk 3]
```diff
@@ -23,7 +23,8 @@ import {
   UUID_REGEX,
   NotFoundError,
   BadParamsError,
-  AlreadyPublishedError
+  AlreadyPublishedError,
+  DependencyMissingError
```
Comment (drtyh2o) [inline packages/mds-policy-author/api.ts]: These two errors are used in a complimentary fashion, but their names don't really convey that. Maybe they could be `DependencyConflictError` (409) and `DependencyMissingError` (404)? Or `AlreadyPublishedError/NotPublishedError`?

File: packages/mds-compliance/tests/api.spec.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-compliance/tests/api.spec.ts:814]: This should be actionable now @janedotx

**Pull request merged**